src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/actions/ActionsSummary.java

```diff
@@ -0,0 +1,85 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.actions;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.ValueColumn;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Action summary that can be persisted in SQL
+ *
+ * <p>Table name : ActionsSummary
+ *
+ * <p>schema :
+ * | id(primary key) |     actionName       |  resourceValue    |  timestamp        |   nodeId      |
+ * |      1          | ModifyQueueCapacity  |      1            |  1599257910923    |   node1       |
+ * | nodeIp     | actionable | coolOffPeriod
+ * | 127.0.0.1  |  true      |  300
+ */
+public class ActionsSummary {
```

Comment:
Actions can be wide ranging and diverse, from resource tuning ones like queues and caches, to cluster management actions like shard movements. We should strive to keep that freedom in our action objects. This is why the `Action` interface was kept generic enough to accommodate for these different kind of actions - only items that each action must have should be imposed there. We should do the same with summaries.

Having an `ActionSummary` class will impose a rigid structure that actions must adhere to. Instead, we can have a simple factory kind of model for generating actions from summaries here:

Each action exposes two methods: 
```java
/* returns action summary as a json string. 
* We can also use jsonElement but string is just 
*/
public String summary(); more generic. The string would still be json. 

// this one is in each action, not in the interface
// Takes action summary as json string and constructs action object from it
public static Action fromSummary(String jsonRepr); 
```

Now each action has its own implementations for these methods, and thus its own impl for serializing and deserializing its contents. The validators can construct an `Action` object from json responses by calling `fromSummary` and validate if these objects are the expected ones.
e.g.
```java
// throws exception if summary is not parse-able to queue action
ModifyQueueCapacityAction action = ModifyQueueCapacityAction.fromSummary(response); 
assertEquals(ModifyQueueCapacityAction.NAME, action.name());
```

Reply:
+1

Maybe we should have separate Action classes for each action type to let them have the freedom to decide on the schema ?

Reply:
I am adding summary to the ActionSummary class (wrapper). Presently all the actions contains this summary. Will create a separate PR for adding fromSummary. 

About creating separate action and allowing them to decide the schema, might be some more thought. I think we start with this and add more fields as use cases increase.